test/integration/controllermanager/seed/extensionscheck/extensionscheck_test.go

```diff
@@ -91,34 +95,17 @@ var _ = Describe("Seed ExtensionsCheck controller tests", func() {
 
 		ci2 = ci1.DeepCopy()
 		ci2.SetGenerateName("foo-2-")
+	})
 
-		for _, controllerInstallation := range []*gardencorev1beta1.ControllerInstallation{ci1, ci2} {
-			Expect(testClient.Create(ctx, controllerInstallation)).To(Succeed(), controllerInstallation.Name+" should be created")
-
-			controllerInstallation.Status = gardencorev1beta1.ControllerInstallationStatus{
-				Conditions: []gardencorev1beta1.Condition{
-					{Type: "Valid", Status: gardencorev1beta1.ConditionTrue},
-					{Type: "Installed", Status: gardencorev1beta1.ConditionTrue},
-					{Type: "Healthy", Status: gardencorev1beta1.ConditionTrue},
-					{Type: "Progressing", Status: gardencorev1beta1.ConditionFalse},
-				},
-			}
-			Expect(testClient.Status().Update(ctx, controllerInstallation)).To(Succeed(), controllerInstallation.Name+" should be updated")
-			log.Info("Created and updated ControllerInstallation for test", "controllerInstallation", client.ObjectKeyFromObject(controllerInstallation))
-
-			By("Wait until ExtensionsReady condition is set to True")
-			Eventually(func(g Gomega) {
-				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)).To(Succeed())
-				g.Expect(seed.Status.Conditions).To(ContainCondition(OfType(gardencorev1beta1.SeedExtensionsReady), WithStatus(gardencorev1beta1.ConditionTrue), WithReason("AllExtensionsReady")))
-			}).Should(Succeed(), "after creation and update of "+controllerInstallation.Name)
-		}
+	JustBeforeEach(func() {
+		createAndUpdateControllerInstallation(ci1, seed, gardencorev1beta1.ConditionFalse)
+		createAndUpdateControllerInstallation(ci2, seed, gardencorev1beta1.ConditionProgressing)
```

Comment:
Seed `ConditionFalse` will happen here also? or not?

Reply:
Nop. It will become `True` after the first `createAndUpdateControllerInstallation(...)` function finishes. When a second `ControllerInstallation` (that has no conditions) is deployed as part of the second function the condition will transition from `True` to `Progressing` as there is a conditionThreshold that has not expired. Then when the second `ControllerInstallation` conditions are updated to be successful, the `ExtensionsReady` will transition from `Progressing` to `True`